src/index.c

```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "index.h"
+
+
+/* Call to create a new empty Index() */
+Index* init_index(int size) {
+	Index *index = (Index*) malloc(sizeof(Index));
+	if (index == NULL) {
+		printf("Failed to allocate memory for index\n");
+		return NULL;
+	}
+
+	// create contents and set all pointers to null
+	Entry **contents = (Entry **) malloc(size * sizeof(Entry*));
+	for (int i = 0; i < size; i++) {
+		*(contents + i) = NULL;
+	}
+
+	if (contents == NULL) {
```

Comment:
This check should be above the for loop.

Reply:
nice catch!